addons/mail/static/src/models/suggested_partners/suggested_partners.js

```diff
@@ -0,0 +1,40 @@
+odoo.define('mail/static/src/models/suggested_partners/suggested_partners.js', function (require) {
+    'use strict';
+
+    const { registerNewModel } = require('mail/static/src/model/model_core.js');
+    const { attr, many2many, many2one } = require('mail/static/src/model/model_field.js');
+    const mailUtils = require('mail.utils');
+
+    function factory(dependencies) {
+        class SuggestedPartner extends dependencies['mail.model'] {
+            init(...args) {
+                super.init(...args);
+            }
+
+            static convertData(data) {
+                const parsedEmail = data[1] && mailUtils.parseEmail(data[1]);
+                const data2 = {
+                    checked: true,
+                    id: data[0],
+                    full_name: data[1],
+                    name: parsedEmail[0],
+                    email_address: parsedEmail[1],
+                    reason: data[2],
+                };
+                return data2;
+            }
+        }
+        SuggestedPartner.fields = {
+            id: attr(),
+            checked: attr({ default: true }),
+            full_name: attr(),
+            name: attr(),
+            email_address: attr(),
+            reason: attr(),
+        };
+        SuggestedPartner.modelName = 'mail.suggested_partner';
```

Comment:
I don't like the model name, it's confusing because would assume that `thread.suggestedPartners` returns `mail.partner`, but instead it's `mail.suggested_partner`, which itself has a relation to `mail.partner`, so code would look something like `thread.suggestedPartners[x].partner`... that's weird...

I think this model should be something like `mail.suggested_partner_info`:
```js
thread.suggestedPartnerInfos
```
That way, it's more natural:
```js
thread.suggestedPartnerInfos[x].partner
thread.suggestedPartnerInfos[x].isChecked
```
Could even have related field, so that access to suggested partners is even more natural:
```js
thread.suggestedPartners; // related to `suggestedPartnerInfos.partner`
```